symengine/fields.h

```diff
@@ -0,0 +1,372 @@
+    {
+        integer_class temp;
+        mp_fdiv_r(temp, a, modulo_);
+        if (temp != integer_class(0)) {
+            auto t = dict_.lower_bound(pos);
+            if (t != dict_.end() and t->first == pos) {
+                t->second = temp;
+            } else {
+                dict_.insert(t, {pos, temp});
+            }
+        }
+    }
+    GaloisFieldDict(GaloisFieldDict &&other) SYMENGINE_NOEXCEPT
+        : ODictWrapper(std::move(other.dict_)), modulo_(std::move(other.modulo_))
+    {
+    }
+    GaloisFieldDict(const int &i, const integer_class &mod) : modulo_(mod)
+    {
+        integer_class temp;
+        mp_fdiv_r(temp, integer_class(i), modulo_);
+        if (temp != integer_class(0)) {
+            dict_.insert({0, temp});
+        }
+    }
+    GaloisFieldDict(const map_uint_mpz &p, const integer_class &mod) : modulo_(mod)
+    {
+        for (auto &iter : p) {
+            integer_class temp;
+            mp_fdiv_r(temp, iter.second, modulo_);
+            if (temp != integer_class(0)) {
+                dict_.insert(dict_.end(), {iter.first, temp});
+            }
+        }
+    }
+    GaloisFieldDict(const integer_class &i, const integer_class &mod) : modulo_(mod)
+    {
+        integer_class temp;
+        mp_fdiv_r(temp, i, modulo_);
+        if (temp != integer_class(0)) {
+            dict_.insert({0, temp});
+        }
+    }
+
+    static GaloisFieldDict from_vec(const std::vector<integer_class> &v, const integer_class &modulo)
+    {
+        GaloisFieldDict x;
+        for (unsigned int i = 0; i < v.size(); ++i) {
+            integer_class a;
+            mp_fdiv_r(a, v[i], modulo);
+            if (a != integer_class(0)) {
+                x.dict_.insert({i, a});
+            }
+        }
+        x.modulo_ = modulo;
+        return x;
+    }
+
+    GaloisFieldDict(const GaloisFieldDict &) = default;
+    GaloisFieldDict &operator=(const GaloisFieldDict &) = default;
+    void gf_div(const GaloisFieldDict &o,
+                const Ptr<GaloisFieldDict> &quo,
+                const Ptr<GaloisFieldDict> &rem) const;
+
+    GaloisFieldDict gf_lshift(const integer_class n) const;
+    void gf_rshift(const integer_class n,
+                   const Ptr<GaloisFieldDict> &quo,
+                   const Ptr<GaloisFieldDict> &rem) const;
+    GaloisFieldDict gf_sqr() const;
+    GaloisFieldDict gf_pow(const integer_class n) const;
+    void gf_monic(integer_class &res, const Ptr<GaloisFieldDict> &monic) const;
+    GaloisFieldDict gf_gcd(const GaloisFieldDict &o) const;
+    GaloisFieldDict gf_lcm(const GaloisFieldDict &o) const;
+
+    GaloisFieldDict &operator=(GaloisFieldDict &&other) SYMENGINE_NOEXCEPT
+    {
+        if (this != &other) {
+            dict_ = std::move(other.dict_);
+            modulo_ = std::move(other.modulo_);
+        }
+        return static_cast<GaloisFieldDict &>(*this);
+    }
+
+    GaloisFieldDict &operator+=(const GaloisFieldDict &other)
```

Comment:
Can you add, `GaloisFieldDict operator+(const GaloisFieldDict &other)` as well? Also `GaloisFieldDict &operator+=(const integer_class &other)`